index.js

```diff
@@ -670,12 +670,22 @@ function nested (laterCode, name, key, schema, externalSchema, fullSchema, subKe
         type.forEach((type, index) => {
           var tempSchema = {type: type}
           var nestedResult = nested(laterCode, name, key, tempSchema, externalSchema, fullSchema, subKey)
-          code += `
-            ${index === 0 ? 'if' : 'else if'}(ajv.validate(${require('util').inspect(tempSchema, {depth: null})}, obj${accessor}))
+          if (type === 'string') {
+            code += `
+              ${index === 0 ? 'if' : 'else if'}(obj${accessor} instanceof Date || ajv.validate(${require('util').inspect(tempSchema, {depth: null})}, obj${accessor}))
+                ${nestedResult.code}
+            `
+          } else {
+            code += `
+              ${index === 0 ? 'if' : 'else if'}(ajv.validate(${require('util').inspect(tempSchema, {depth: null})}, obj${accessor}))
               ${nestedResult.code}
-          `
```

Comment:
Why is this call to `ajv` needed? I would refrain from doing so. Callling ajv is costly.

Reply:
@mcollina I'm not sure really. All I have done is to check if the data is a Date when the type is string. the ajv stuff was there before already.